node/src/accountant/scanners_tools.rs

```diff
@@ -0,0 +1,539 @@
+    };
+    use crate::accountant::scanners_tools::receivable_scanner_tools::balance_and_age;
+    use crate::accountant::SentPayable;
+    use crate::blockchain::blockchain_interface::BlockchainError;
+    use crate::database::dao_utils::{from_time_t, to_time_t};
+    use crate::sub_lib::accountant::PaymentThresholds;
+    use crate::test_utils::make_wallet;
+    use masq_lib::logger::Logger;
+    use std::rc::Rc;
+    use std::time::SystemTime;
+
+    #[test]
+    fn payable_generated_before_maturity_time_limit_is_marked_unqualified() {
+        let now = SystemTime::now();
+        let payment_thresholds = PaymentThresholds::default();
+        let qualified_debt = payment_thresholds.permanent_debt_allowed_gwei + 1;
+        let unqualified_time = to_time_t(now) - payment_thresholds.maturity_threshold_sec + 1;
+        let unqualified_payable_account = PayableAccount {
+            wallet: make_wallet("wallet0"),
+            balance: qualified_debt,
+            last_paid_timestamp: from_time_t(unqualified_time),
+            pending_payable_opt: None,
+        };
+
+        let result = is_payable_qualified(now, &unqualified_payable_account, &payment_thresholds);
+
+        assert_eq!(result, None);
+    }
+
+    #[test]
+    fn payable_with_low_debt_is_marked_unqualified() {
+        let now = SystemTime::now();
+        let payment_thresholds = PaymentThresholds::default();
+        let unqualified_debt = payment_thresholds.permanent_debt_allowed_gwei - 1;
+        let qualified_time = to_time_t(now) - payment_thresholds.maturity_threshold_sec - 1;
+        let unqualified_payable_account = PayableAccount {
+            wallet: make_wallet("wallet0"),
+            balance: unqualified_debt,
+            last_paid_timestamp: from_time_t(qualified_time),
+            pending_payable_opt: None,
+        };
+
+        let result = is_payable_qualified(now, &unqualified_payable_account, &payment_thresholds);
+
+        assert_eq!(result, None);
+    }
+
+    #[test]
+    fn payable_with_low_payout_threshold_is_marked_unqualified() {
+        let now = SystemTime::now();
+        let payment_thresholds = PaymentThresholds::default();
+        let debt = payment_thresholds.permanent_debt_allowed_gwei + 1;
+        let time = to_time_t(now) - payment_thresholds.maturity_threshold_sec - 1;
+        let unqualified_payable_account = PayableAccount {
+            wallet: make_wallet("wallet0"),
+            balance: debt,
+            last_paid_timestamp: from_time_t(time),
+            pending_payable_opt: None,
+        };
+
+        let result = is_payable_qualified(now, &unqualified_payable_account, &payment_thresholds);
+
+        assert_eq!(result, None);
+    }
+
+    #[test]
+    fn payable_above_threshold_values_is_marked_qualified_and_returns_threshold() {
+        let now = SystemTime::now();
+        let payment_thresholds = PaymentThresholds::default();
+        let debt = payment_thresholds.permanent_debt_allowed_gwei + 1_000_000_000;
```

Comment:
Not a good enough proof. You need to be more brave in approaching the critical area. Also, the constant 1_000_000_000 is a bad idea because your `payment_thresholds` derives from the defaults, which can change over time flexibly to the market situation and the team reconsideration. 

This should be more like    
```
let debt = payment_thresholds.debt_threshold - 1;   //1 .. or even more
let time = payment_thresholds.maturity_threshold_sec + payment_threshold.threshold_interval_sec - 1; 
//1 ..or better if more  
```